cli_tools/daisy_test_runner/main.go

```diff
@@ -351,7 +367,125 @@ type test struct {
 	testCase *TestCase
 }
 
-func runTestCase(ctx context.Context, test *test, tc *junitTestCase, errors chan error) {
+func getCommonInstanceMetadata(client daisyCompute.Client, project string) (*compute.Metadata, error) {
+	proj, err := client.GetProject(project)
+	if err != nil {
+		return nil, fmt.Errorf("error getting project: %v", err)
+	}
+
+	return proj.CommonInstanceMetadata, nil
+}
+
+func delItem(items []*compute.MetadataItems, i int) []*compute.MetadataItems {
+	// Delete the element.
+	// https://github.com/golang/go/wiki/SliceTricks
+	copy(items[i:], items[i+1:])
+	items[len(items)-1] = nil
+	return items[:len(items)-1]
+}
+
+func isExpired(val string) bool {
+	t, err := time.Parse(time.RFC3339, val)
+	if err != nil {
+		return false
+	}
+	return time.Now().After(t)
+}
+
+const (
+	writeLock      = "TestWriteLock-"
+	readLock       = "TestReadLock-"
+	defaultTimeout = 2 * time.Hour
+)
+
+func waitLock(client daisyCompute.Client, project, prefix string) (*compute.Metadata, error) {
+	locked := true
+	var md *compute.Metadata
+	var err error
+	for locked {
+		locked = false
+		md, err = getCommonInstanceMetadata(client, project)
+		if err != nil {
+			return nil, err
+		}
+
+		for i, mdi := range md.Items {
+			if mdi != nil && strings.HasPrefix(mdi.Key, prefix) {
+				if isExpired(*mdi.Value) {
+					md.Items = delItem(md.Items, i)
+				} else {
+					locked = true
+					r := rand.Intn(10) + 5
```

Comment:
minor: it might be helpful to add a small comment. eg. "Sleep for a random interval between 5 and 15 seconds before checking again."